bin/docserv-createconfig

```diff
@@ -220,9 +222,66 @@ def get_dc_files(path, branches, filter_dc=None):
     tmp = branches
     with switchbranch(repo) as checkout_branch:
         for branch in checkout_branch(branches):
-            tmp[branch] = collect_dc_files(path, filter_dc)
+            path2git = path + "/en" if "en" in os.listdir(path) else path
+            tmp[branch] = collect_dc_files(path2git, filter_dc)
     return tmp
 
+def get_subdeliverables(path, dc_files):
+    """Collect all subdeliverables from DC-*-all files in all branches.
+
+    :param str path: the path to the Git repository
+    :param dict dc_files: A dictionary with lists as keys like
+         {'branch1': [], 'branch2': [], ...}
+    :return: a dictionary with all subdeliverables found in each branch with
+         lists as keys like {'branch1DC-file1-all': [], 'branch2DC-file2-all': [], ...}
+    :rtype: dict
+    """
+    result = {}
+    repo = pygit2.Repository(path)
+    with switchbranch(repo) as checkout_branch:
+        for branch in checkout_branch(branches):
+            path2git = path + "/en" if "en" in os.listdir(path) else path
+            for dc_file in dc_files[branch]:
+                if dc_file.endswith("-all"):
+                    result[branch + dc_file] = collect_subdeliverables(path2git, dc_file)
+    return result
+
+def collect_subdeliverables(path, dc_file):
+    """Collect all subdeliverables from given DC-file
+
+    :param str path: the path to the Git repository
+    :param str dc_file: the DC-file to look into
+    :return: a list with all subdeliverables found after looking in this DC-file
+    :rtype: list
+    """
+    result = []
+    main = ""
+    # example: ('MAIN', '=', '"', 'MAIN.susestorage.xml', '"')
+    #        ('ROOTID', '=', None, 'book.storage.admin', None)
+    regex = r'([A-Z]+)(=)("|\')?([\w\.\-]+)("|\')?'
+    with open(os.path.join(path, dc_file), "r") as f:
+        for line in f:
+            match = re.match(regex, line)
+            if match:
+                if match.group(1) == "MAIN":
+                    main = match.group(4)
+    if main:
+        try:
+            root = etree.parse(os.path.join(path + "/xml", main))
+            namespaces = dict(d="http://docbook.org/ns/docbook", xi="http://www.w3.org/2001/XInclude")
+            xpaths = "/set/xi:include|/set/book/xi:include|" \
+                     "/d:set/xi:include|/d:set/d:book/xi:include"
```

Comment:
I have a suggestion that will make some significant work but should be the better option: Look at the _other_ DC files in the repo instead of running all this XML stuff: For everything we publish individually, we usually have an own DC file which basically references a subset of the `DC-*-all` document. Here's what came to my mind:

1. Check the `MAIN=` value in the `DC-*-all` file
2. Check all of the other DC files in the repository
   * Is the MAIN value different? They are irrelevant
   * Is the MAIN value the same? Collect the ROOTID value
3. Use all of the collected ROOTIDs as subdeliverables for your `DC-*-all`

This way, you are also catching those articles that you had a hard time getting at so far. You can also avoid parsing all this XML.

Reply:
Your suggestion is much simpler and easier, although I'm not sure which one of these two solution would be best:
 * one checks thoroughly, by parsing XML
 * the other checks marginally, trusting that the DC-files are correct (which ideally it should be correct, but not always: [see sleha](https://github.com/SUSE/doc-sleha/pull/105))

Reply:
Created new PR #144